### PR TITLE
Cross cloud support

### PR DESCRIPTION
**Pull request by jasoncoolmax**
## Proposed changes

The main change is to make sure we only compare `MSALAccount` based on home account id regardless of authority. Other logic to fill up the tenant profile is already in place.

Common core changes: https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/892

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: MSAL/IdentityCore [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 525cfb35eb7ea82ca6b9e1cfc1af2bf403bbf39f
```
Comment (kaisong1990) [inline MSAL/IdentityCore:1]: This pull request does not update CHANGELOG.md.

Please consider if this change would be noticeable to a partner or user and either update CHANGELOG.md or resolve this conversation.

**Review by kaisong1990 — APPROVED** [review]

File: MSAL/src/MSALAccount.m [hunk 2]
```diff
@@ -161,7 +161,8 @@ - (BOOL)isEqual:(id)object
 - (NSUInteger)hash
 {
     NSUInteger hash = 0;
-    hash = hash * 31 + self.environment.hash;
+    // Equality of MSALAccount is depending on equality of homeAccountId or username
+    // So we are not able to calculate a precise hash
     return hash;
 }
```
Comment (hieunguyenmsft) [inline MSAL/src/MSALAccount.m:167]: Should we remove the hash function here? Fix the hash value to 0 seems incorrect in general.

Comment (jasoncoolmax) [reply MSAL/src/MSALAccount.m:167]: `hash` is required if we want to define our own `isEqual` as indicated [here](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418795-isequal).
And it also requires [here](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418859-hash):
> If two objects are equal (as determined by the isEqual: method), they must have the same hash value.

So I think we may need to keep it anyway even though it doesn't help with performance :P

Comment (hieunguyenmsft) [reply MSAL/src/MSALAccount.m:167]: Thank for the info, I see we implement the own logic in `isEqual` so we don't need the hash function other than for the procedure :D

**Pull request merged**